src/Console/Shell.php

```diff
@@ -549,6 +548,8 @@ protected function _displayHelp($command)
             $this->_welcome();
         }
 
+        $subcommands = $this->OptionParser->subcommands();
+        $command = isset($subcommands[$command]) ? $command : null;
```

Comment:
This is necessary because unknown command will now trigger the new error message.
However, when you type `bin/cake plugin -h`, `-h` will be considered as a subcommand (but the parser will notice that the help is required, which is why we end up in the `_displayHelp()` method in the first place).

This change allows the `ConsoleOptionParser::help()` to display the good help message : if the subcommand exists, we'll display the subcommand's help, if not, we display the command general help and not the new error message.